### PR TITLE
Add instantly option to deepSleep

### PR DESCRIPTION
**Pull request by torntrousers**
As mentioned in https://github.com/esp8266/Arduino/issues/3408#issuecomment-413279368 here's an update to add an 'instantly' option to the deepSleep function.

### CODE REVIEW

File: cores/esp8266/Esp.h [hunk 1]
```diff
@@ -92,7 +92,7 @@ class EspClass {
         void wdtDisable();
         void wdtFeed();
 
-        void deepSleep(uint64_t time_us, RFMode mode = RF_DEFAULT);
+        void deepSleep(uint64_t time_us, RFMode mode = RF_DEFAULT, bool instantly = false);
```
Comment (igrr) [inline cores/esp8266/Esp.h]: Suggest adding new function, `deepSleepInstant`. This way, only one of `system_deep_sleep_instant`/`system_deep_sleep` will be linked into the final application, depending on what the user needs.

Comment (torntrousers) [reply cores/esp8266/Esp.h]: Ok, done.

**Comment by 5chufti** [comment]
together with the request for a safe max option like requested in #4969 I opt for something like proposed here https://github.com/esp8266/Arduino/pull/4936#issuecomment-410435875

**Comment by torntrousers** [comment]
Doesn't that still have the issue that @igrr just mentioned of linking in a bit more than necessary. Also ```0x1800000000ull``` is a little obscure compared to explicitly mentioning `instant`` somewhere.

**Comment by 5chufti** [comment]
yes but then why not also seperate ESP.deepSleep() and ESP.setWakeUp()
to me this is academic hair-splitting. To get least pulled in from SDK, one should not combine any SDK functions within single ESP.class method at all, ending up in ESP.class just passing through simple SDK api. With the suggested workaround for #3408 it's even one more ...

I don't mind having 2 additional bools, 
> void deepSleep(uint64_t time_us, RFMode mode = RF_DEFAULT, bool max_safe = FALSE, bool instant = FALSE);

no hard feelings for obscure flags in arguments, just wanted to keep it max compat.

**Comment by igrr** [comment]
Dear @5chufti, if a user encounters the following piece of code

```c++
ESP.deepSleep(duration, RF_DEFAULT, true);
```

they will need to go find the function declaration to figure out what "true" means.

Compare this to

```c++
ESP.deepSleepInstant(duration, RF_DEFAULT);
```

where the method name conveys this information.

Admittedly, the core has accumulated numerous APIs with optional boolean arguments, most via simple backwards-compatible enhancements like this one (in the original form of this PR). WiFi class is a good example. It doesn't mean though that the problem should be ignored.

With regards to pulling in SDK functions — this is not academic hair splitting. Let me give you an extreme example. In the first version of HTTP Client library, there was code like this:
```c++
// parse URL string, get host, port, protocol, URI
if (protocol == "https") {
    client = new WiFiClientSecure(host, port);
} else {
    client = new WiFiClient(host, port);
}
```

This is an example of bad design. User usually only needs A or B, but the code includes functionality for both A and B at link time. In this specific case, it pulled in *lots* of SSL-related code, greatly increasing the binary size.

It's going to be less extreme with system_deep_sleep, but still, this is a thing to consider.

**Comment by 5chufti** [comment]
@igrr 
> ESP.deepSleepInstant(duration, RF_DEFAULT);
sorry, but here the user has to find the useable definitions that are not even the same as in SDK !

**Comment by torntrousers** [comment]
Cheers @igrr . Don't suppose you could also pop over [here](https://github.com/esp8266/Arduino/issues/3408#issuecomment-355725812) and comment again on that issue?

**Comment by igrr** [comment]
WiFi is turned off as well, just not gracefully.

**Comment by d-a-v** [comment]
OK, I was mistaken by "false friend" `instant` word between french and english.

**Pull request merged**